### DISCUSSION TITLE
Recipes are gone

### DISCUSSION THREAD

**Discussion by SteveDevDev** [mealie-recipes/mealie#1004] [Q&A]
I have been running Mealie using Docker Compose for a while.  Everything worked fine.  I decided to run "docker-compose pull" to update Mealie.  That downloaded and updated fine, but when Mealie came back up all of the recipes, categories, tags are gone.  It still remembers my dark mode settings.  I figured Postgres had some sort of problem but I don't see anything in the logs.  According to the logs there are no errors with the mealie containor nor the postgres container.  

How can I troubleshoot this/

**Comment by SteveDevDev** [comment]
Here is the docker-compose.yml file:

`version: "3.1"
services:
  mealie:
    build:
      context: ./
      dockerfile: Dockerfile
    container_name: mealie
    restart: always
    depends_on:
      - "postgres"
    ports:
      - 8089:80
    environment:
      DB_ENGINE: postgres # Optional: 'sqlite', 'postgres'
      DEFAULT_EMAIL: changeme@email.com
      POSTGRES_USER: mealie
      POSTGRES_PASSWORD: mealie
      POSTGRES_SERVER: postgres
      POSTGRES_PORT: 5432
      POSTGRES_DB: mealie
  postgres:
    container_name: postgres
    image: postgres
    restart: always
    environment:
      POSTGRES_PASSWORD: mealie
      POSTGRES_USER: mealie`

**Comment by hay-kot** [comment]
Hey! Depending on what version you upgraded from it's likely the database schema has changed. Currently we don't provide automatic migrations. As such, upgrade should be done with caution and reading the change-log. Easiest thing to do would be to roll-back to the last version you were on, performance an export. Upgrade, and then re-import your data. 

We will have the automatic-migrations setup once we move to v1 - which is about to enter it's beta testing phase.

**Reply by SteveDevDev** [reply]
oh I see.  What version do you recommend running?  It looks like the image I'm on and the one before it is Postgres 14.  When I roll back to Postgres 13 I get this error:

`PostgreSQL Database directory appears to contain a database; Skipping initialization
2022-02-24 23:12:00.809 UTC [1] FATAL:  database files are incompatible with server
2022-02-24 23:12:00.809 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 14, which is not compatible with this version 13.5 (Debian 13.5-1.pgdg110+1).`

Do you think a tag in the docker-compose.yml to specify the correct image would have prevented this?  I'm thinking not specifying an image tag means it will pull down the latest.

Thanks

**Reply by hay-kot** [reply]
You don't need to roll back PostgreSQL, you need to roll back your Mealie version. 

> Do you think a tag in the docker-compose.yml to specify the correct image would have prevented this?

Yes, this is why, generally, it's a good idea to pin your version and monitor version updates. I'm guilty of using :latest and updating though, no shame in it.

**Reply by SteveDevDev** [reply]
Hmm.. it doesn't look like the mealie image has been updated in a while.  This is the only one in my "docker image list"

mealie_mealie                  latest            2f04f820c956   8 months ago    539MB

whereas Postgres has been updated several times (some of the tags are mine)

postgres                       latest            6a3c44872108   10 days ago     374MB
postgres                       working           da2cb49d7a8d   4 weeks ago     374MB
postgres                       13                0896a8e0282d   7 weeks ago     371MB
postgres                       <none>            7975f8e74a95   3 months ago    371MB
postgres                       <none>            b2fcd079c1d4   7 months ago    315MB
postgres                       <none>            ed45d5bb6847   8 months ago    315MB

**Comment by SteveDevDev** [comment]
Can anyone help?  Here is my Mealie log.  It doesn't indicate any problems, but the recipes are gone.

    purge_events_database (trigger: interval[1 day, 0:00:00], next run at: 2022-03-21 01:34:36 UTC)
INFO: 20-Mar-22 05:34:36        None
INFO: 20-Mar-22 05:34:36        Job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 06:04:36 UTC)" executed successfully
INFO: 20-Mar-22 06:04:36        Running job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 06:34:36 UTC)" (scheduled at 2022-03-20 06:04:36.826081+00:00)
Jobstore default:
    update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 06:34:36 UTC)
    backups (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    Home (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    purge_events_database (trigger: interval[1 day, 0:00:00], next run at: 2022-03-21 01:34:36 UTC)
INFO: 20-Mar-22 06:04:36        None
INFO: 20-Mar-22 06:04:36        Job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 06:34:36 UTC)" executed successfully
INFO: 20-Mar-22 06:34:36        Running job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 07:04:36 UTC)" (scheduled at 2022-03-20 06:34:36.826081+00:00)
Jobstore default:
    update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 07:04:36 UTC)
    backups (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    Home (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    purge_events_database (trigger: interval[1 day, 0:00:00], next run at: 2022-03-21 01:34:36 UTC)
INFO: 20-Mar-22 06:34:36        None
INFO: 20-Mar-22 06:34:36        Job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 07:04:36 UTC)" executed successfully
INFO: 20-Mar-22 07:04:36        Running job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 07:34:36 UTC)" (scheduled at 2022-03-20 07:04:36.826081+00:00)
Jobstore default:
    update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 07:34:36 UTC)
    backups (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
INFO: 20-Mar-22 07:04:36        Job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 07:34:36 UTC)" executed successfully
    Home (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    purge_events_database (trigger: interval[1 day, 0:00:00], next run at: 2022-03-21 01:34:36 UTC)
INFO: 20-Mar-22 07:04:36        None
INFO: 20-Mar-22 07:34:36        Running job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 08:04:36 UTC)" (scheduled at 2022-03-20 07:34:36.826081+00:00)
Jobstore default:
    update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 08:04:36 UTC)
    backups (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    Home (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    purge_events_database (trigger: interval[1 day, 0:00:00], next run at: 2022-03-21 01:34:36 UTC)
INFO: 20-Mar-22 07:34:36        None
INFO: 20-Mar-22 07:34:36        Job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 08:04:36 UTC)" executed successfully
INFO: 20-Mar-22 08:04:36        Running job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 08:34:36 UTC)" (scheduled at 2022-03-20 08:04:36.826081+00:00)
Jobstore default:
    update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 08:34:36 UTC)
    backups (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    Home (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    purge_events_database (trigger: interval[1 day, 0:00:00], next run at: 2022-03-21 01:34:36 UTC)
INFO: 20-Mar-22 08:04:36        None
INFO: 20-Mar-22 08:04:36        Job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 08:34:36 UTC)" executed successfully
INFO: 20-Mar-22 08:34:36        Running job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 09:04:36 UTC)" (scheduled at 2022-03-20 08:34:36.826081+00:00)
Jobstore default:
    update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 09:04:36 UTC)
    backups (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    Home (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    purge_events_database (trigger: interval[1 day, 0:00:00], next run at: 2022-03-21 01:34:36 UTC)
INFO: 20-Mar-22 08:34:36        None
INFO: 20-Mar-22 08:34:36        Job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 09:04:36 UTC)" executed successfully
INFO: 20-Mar-22 09:04:36        Running job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 09:34:36 UTC)" (scheduled at 2022-03-20 09:04:36.826081+00:00)
Jobstore default:
    update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 09:34:36 UTC)
    backups (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    Home (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    purge_events_database (trigger: interval[1 day, 0:00:00], next run at: 2022-03-21 01:34:36 UTC)
INFO: 20-Mar-22 09:04:36        None
INFO: 20-Mar-22 09:04:36        Job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 09:34:36 UTC)" executed successfully
INFO: 20-Mar-22 09:34:36        Running job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 10:04:36 UTC)" (scheduled at 2022-03-20 09:34:36.826081+00:00)
Jobstore default:
    update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 10:04:36 UTC)
    backups (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    Home (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    purge_events_database (trigger: interval[1 day, 0:00:00], next run at: 2022-03-21 01:34:36 UTC)
INFO: 20-Mar-22 09:34:36        None
INFO: 20-Mar-22 09:34:36        Job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 10:04:36 UTC)" executed successfully
INFO: 20-Mar-22 10:04:36        Running job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 10:34:36 UTC)" (scheduled at 2022-03-20 10:04:36.826081+00:00)
Jobstore default:
    update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 10:34:36 UTC)
    backups (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    Home (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    purge_events_database (trigger: interval[1 day, 0:00:00], next run at: 2022-03-21 01:34:36 UTC)
INFO: 20-Mar-22 10:04:36        None
INFO: 20-Mar-22 10:04:36        Job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 10:34:36 UTC)" executed successfully
INFO: 20-Mar-22 10:34:36        Running job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 11:04:36 UTC)" (scheduled at 2022-03-20 10:34:36.826081+00:00)
Jobstore default:
    update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 11:04:36 UTC)
    backups (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    Home (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    purge_events_database (trigger: interval[1 day, 0:00:00], next run at: 2022-03-21 01:34:36 UTC)
INFO: 20-Mar-22 10:34:36        None
INFO: 20-Mar-22 10:34:36        Job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 11:04:36 UTC)" executed successfully
INFO:     172.31.0.1:0 - "GET /api/debug/version HTTP/1.1" 200 OK
INFO:     172.31.0.1:0 - "GET /api/recipes/summary HTTP/1.1" 200 OK
INFO:     172.31.0.1:0 - "GET /api/tags HTTP/1.1" 200 OK
INFO:     172.31.0.1:0 - "GET /api/groups/self HTTP/1.1" 401 Unauthorized
INFO:     172.31.0.1:0 - "GET /api/categories HTTP/1.1" 200 OK
INFO:     172.31.0.1:0 - "GET /api/site-settings HTTP/1.1" 200 OK
INFO:     172.31.0.1:0 - "GET /api/recipes/summary HTTP/1.1" 200 OK
INFO:     172.31.0.1:0 - "GET /api/site-settings/custom-pages HTTP/1.1" 200 OK
INFO:     172.31.0.1:0 - "GET /api/site-settings HTTP/1.1" 200 OK
INFO: 20-Mar-22 11:04:36        Running job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 11:34:36 UTC)" (scheduled at 2022-03-20 11:04:36.826081+00:00)
Jobstore default:
    update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 11:34:36 UTC)
    backups (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    Home (trigger: cron[hour='0', minute='0'], next run at: 2022-03-21 00:00:00 UTC)
    purge_events_database (trigger: interval[1 day, 0:00:00], next run at: 2022-03-21 01:34:36 UTC)
INFO: 20-Mar-22 11:04:36        None
INFO: 20-Mar-22 11:04:36        Job "update_webhook_schedule (trigger: interval[0:30:00], next run at: 2022-03-20 11:34:36 UTC)" executed successfully
Skipping virtualenv creation, as specified in config file.
Database Exists
Skipping virtualenv creation, as specified in config file.
INFO: 20-Mar-22 11:06:17        Checking for Images to Minify...
INFO: 20-Mar-22 11:06:17        Finished Minification Check
Production
{"level":"info","ts":1647774378.154704,"msg":"using provided configuration","config_file":"./Caddyfile","config_adapter":""}
{"level":"warn","ts":1647774378.1578097,"msg":"input is not formatted with 'caddy fmt'","adapter":"caddyfile","file":"./Caddyfile","line":3}
{"level":"warn","ts":1647774378.158528,"logger":"admin","msg":"admin endpoint disabled"}
{"level":"info","ts":1647774378.1600893,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0xc0002e27e0"}
{"level":"info","ts":1647774378.1687262,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/root/.local/share/caddy"}
{"level":"info","ts":1647774378.168772,"logger":"tls","msg":"finished cleaning storage units"}
{"level":"info","ts":1647774378.1707432,"msg":"autosaved config (load with --resume flag)","file":"/root/.config/caddy/autosave.json"}
{"level":"info","ts":1647774378.170783,"msg":"serving initial configuration"}
Successfully started Caddy (pid=21) - Caddy is running in the background
INFO:     Started server process [33]
INFO: 20-Mar-22 11:06:24        Started server process [33]
INFO:     Waiting for application startup.
INFO: 20-Mar-22 11:06:24        Waiting for application startup.
INFO: 20-Mar-22 11:06:24        Adding job tentatively -- it will be properly scheduled when the scheduler starts
INFO: 20-Mar-22 11:06:24        Adding job tentatively -- it will be properly scheduled when the scheduler starts
INFO: 20-Mar-22 11:06:24        ----INIT SCHEDULE OBJECT-----
INFO: 20-Mar-22 11:06:24        Adding job tentatively -- it will be properly scheduled when the scheduler starts
INFO: 20-Mar-22 11:06:25        Adding job tentatively -- it will be properly scheduled when the scheduler starts
Pending jobs:
    purge_events_database (trigger: interval[1 day, 0:00:00], pending)
    update_webhook_schedule (trigger: interval[0:30:00], pending)
    backups (trigger: cron[hour='0', minute='0'], pending)
    Home (trigger: cron[hour='0', minute='0'], pending)
INFO: 20-Mar-22 11:06:25        None
INFO: 20-Mar-22 11:06:25        Added job "purge_events_database" to job store "default"
INFO: 20-Mar-22 11:06:25        Added job "update_webhook_schedule" to job store "default"
INFO: 20-Mar-22 11:06:25        Added job "backups" to job store "default"
INFO: 20-Mar-22 11:06:25        Added job "Home" to job store "default"
INFO: 20-Mar-22 11:06:25        Scheduler started
INFO: 20-Mar-22 11:06:25        -----SYSTEM STARTUP-----

INFO: 20-Mar-22 11:06:25        ------APP SETTINGS------
INFO: 20-Mar-22 11:06:25        {
    "PRODUCTION": true,
    "BASE_URL": "http://localhost:8080",
    "IS_DEMO": false,
    "API_PORT": 9000,
    "API_DOCS": true,
    "DB_ENGINE": "postgres",
    "POSTGRES_USER": "mealie",
    "POSTGRES_PASSWORD": "password",
    "POSTGRES_SERVER": "postgres",
    "POSTGRES_PORT": "5432",
    "POSTGRES_DB": "mealie",
    "DB_URL": "postgresql://mealie:password@postgres:5432/mealie",
    "DEFAULT_GROUP": "Home",
    "DEFAULT_EMAIL": "changeme@email.com",
    "TOKEN_TIME": 2
}
INFO:     Application startup complete.
INFO: 20-Mar-22 11:06:25        Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:9000 (Press CTRL+C to quit)
INFO: 20-Mar-22 11:06:25        Uvicorn running on http://0.0.0.0:9000 (Press CTRL+C to quit)
None
INFO:     172.16.10.210:0 - "GET /api/recipes/summary HTTP/1.1" 200 OK
INFO:     172.16.10.210:0 - "GET /api/groups/self HTTP/1.1" 401 Unauthorized
INFO:     172.16.10.210:0 - "GET /api/site-settings HTTP/1.1" 200 OK
INFO:     172.16.10.210:0 - "GET /api/debug/version HTTP/1.1" 200 OK
INFO:     172.16.10.210:0 - "GET /api/categories HTTP/1.1" 200 OK
INFO:     172.16.10.210:0 - "GET /api/tags HTTP/1.1" 200 OK
INFO:     172.16.10.210:0 - "GET /api/themes/1 HTTP/1.1" 200 OK
INFO:     172.16.10.210:0 - "GET /api/site-settings HTTP/1.1" 200 OK
INFO:     172.16.10.210:0 - "GET /api/site-settings/custom-pages HTTP/1.1" 200 OK
INFO:     172.16.10.210:0 - "GET /api/recipes/summary HTTP/1.1" 200 OK

**Comment by SteveDevDev** [comment]
Seeing this.  Anyway to rollback the database or upgrade mealie to be compatible?

postgres    |
postgres    | PostgreSQL Database directory appears to contain a database; Skipping initialization
postgres    |
postgres    | 2022-03-20 11:27:21.447 UTC [1] FATAL:  database files are incompatible with server
postgres    | 2022-03-20 11:27:21.447 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 13, which is not compatible with this version 14.1 (Debian 14.1-1.pgdg110+1).
postgres exited with code 1

**Comment by SteveDevDev** [comment]
Anyway to get support here?

**Reply by hay-kot** [reply]
I'm not sure what else I can offer here. It doesn't look like the error has anything to do with Mealie, but that you've auto-updated your PostgreSQL database. More over, the docs explicitly state that PostgreSQL is in BETA and you should backup your data frequently. 

I suggest that you validate that your data is even present in the Postgres instance anywhere and if it is, I'm not sure what else you can do besides manually export and try to restore.

**Comment by michael-genson** [comment]
Mealie has changed drastically since this discussion was opened, so I'm going ahead and closing this. If you're still having trouble migrating to the newest version, feel free to open an issue or reach out on Discord
